### ISSUE TITLE
How to start porting to mainline

### ISSUE THREAD

**Issue by giuliobenetti** [free-electrons/linux-cedrus#1]
Hello everybody,

@mripard @tpetazzoni

I'm starting to port the patchset for cedrus to mainline.
I have forked mainline linux to https://github.com/micronovasrl/linux-cedrus
Now, how can I proceed?
All patches are obviously full of hunks,
so I can't simply do "git am *.patch".
How can I proceed with commits?
Do I fix patch per patch and reuse "old" commit log?
But I don't think it's right, because there is also porting part to add to commit.
So I don't know how to proceed,
also against this repo with mine.

You can check first commit with previous commit log:
https://github.com/micronovasrl/linux-cedrus/commit/6b20076505630f39799af03d59974e5d9e10b625

Can you provide me some hint to proceed?

Thank you

**Comment by giuliobenetti** [comment]
Also I forked from torvalds.
Maybe I have to fork from linux-sunxi.

After applying all 55 patches where I have to send Pull Request?

**Comment by mripard** [comment]
We already discussed that here:
https://www.spinics.net/lists/linux-media/msg124836.html

tl; dr: Until the request API is merged, we can't mainline anything.

**Comment by giuliobenetti** [comment]
Ok, now it's clear to me.
In the meanwhile I've seen on mailing list that Thomas already applied all patches and went further.
https://github.com/thomas-vitsch/linux-a20-cedrus/commits/linux-sunxi-cedrus

Thank you
